### PR TITLE
[tpcds](nereids) estimate distribution cost by byte size instead of row count

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
this pr impacts tpch q16 Agg strategy, but no performance issue
this pr improves tpcds sf100
before: 
cold 141 sec
hot 133 sec 

after:
code 137 sec
hot 128 sec
Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java [hunk 1]
```diff
@@ -161,27 +161,27 @@ public Cost visitPhysicalPartitionTopN(PhysicalPartitionTopN<? extends Plan> par
...
         DistributionSpec spec = distribute.getDistributionSpec();
+        int beNumber = ConnectContext.get().getEnv().getClusterInfo().getBackendsNumber(true);
+        beNumber = Math.max(1, beNumber);
+        double dataSize = childStatistics.computeSize() / kBytes; // in K bytes
         // shuffle
         if (spec instanceof DistributionSpecHash) {
             return CostV1.of(
                     0,
                     0,
-                    childStatistics.getRowCount());
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:171]: In cost model v1, we use row count as the unified metrics to measure the cost. So here we would better to also use row-count based cost system, if we need to distinguish the distribute cost for different cases, we can use a factor parameter using dataSize as the input with a minimal value 1.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java [hunk 2]
```diff
@@ -191,7 +191,7 @@ public Cost visitPhysicalDistribute(
             return CostV1.of(
                     0,
                     0,
-                    childStatistics.getRowCount() * Math.pow(beNumber, 0.5));
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:194]: In cost model v1, we use row count as the unified metrics to measure the cost. So here we would better to also use row-count based cost system, if we need to distinguish the distribute cost for different cases, we can use a factor parameter using dataSize as the input with a minimal value 1.

**Review by xzj7019 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java (hunk 1 above)
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:171]: ok, let's do it in next pr

**Pull request merged**